metacat-common/src/main/java/com/netflix/metacat/common/MetacatRequestContext.java

```diff
@@ -65,18 +73,21 @@
      * @param clientId        client id
      * @param jobId           job id
      * @param dataTypeContext data type context
+     * @param apiPathPrefix   the path prefix of rest api
      */
     public MetacatRequestContext(
         @Nullable final String userName,
         @Nullable final String clientAppName,
         @Nullable final String clientId,
         @Nullable final String jobId,
-        @Nullable final String dataTypeContext
+        @Nullable final String dataTypeContext,
+        @Nullable final String apiPathPrefix
```

Comment:
This should just be API version and it should probably be an integer. If you intend to capture something like `thrift` vs. `http` then you're talking about some sort of invocation scheme. I'd recommend breaking it up into two fields for clarity to save parsing later on as more API versions or layers cold be added.